test/release/examples/primers/modules.chpl

```diff
@@ -778,19 +817,46 @@ module UsesTheUser2 {
   writeln("Start of UsesTheUser2's module-level code");
   func2();
   {
-    use UsesTheImporter;
+    use UsesTheUser3;
   }
   writeln("End of UsesTheUser2's module-level code");
 }
 
+/* In addition, ``public use`` statements `re-export` the symbols in the used
+   module.  This means that the used symbols can also be treated as though they
+   are defined in the scope with the ``use`` for the purpose of accesses from
+   outside that module.  For example, here it will also appear as though the
+   module ``ModuleThatIsUsed`` is a submodule of ``UserModule2``, so other
```

Comment:
This module name makes my brain hurt, somehow. Could these module names be like "Library" and "BasicLibrary" or even better, make up an example to do with sprockets or linear algebra or something concrete?

Reply:
The trouble is that there are 22 module declarations in this file and we already have one module named to indicate that it is intended to be used in the examples (`modToUse`).  I chose `ModuleThatIsUsed` here over `UsedModule` because the latter would be visually much harder to distinguish from `UserModule`.

Linear algebra is really difficult to distill into a simple example.  Naming them after the concept being explained will make it less likely someone breaks the intended purpose in the future if something in the primer needs to be updated.  We also want these modules to be defined near where they are used so that readers can easily flip between the involved code, instead of having the user have to know what symbols are defined in a standard or package module for instance.